### PR TITLE
Fixes #17994: add a backtrace option for error handling

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17994

### CODE REVIEW

**Comment by gpoblon** [comment]
implementation + backtrace for panics but no backtrace for std errors yet. 
The backtrace can be called + created with 1 line, from wherever in the code for debug purposes

File: rudder-lang/src/logger.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // SPDX-FileCopyrightText: 2019-2020 Normation SAS
 
 use crate::Action;
+use backtrace::{ Backtrace, BacktraceSymbol };
```
Comment (amousset) [inline rudder-lang/src/logger.rs:5]: does not look like `rustfmt` was run on this

Comment (gpoblon) [reply rudder-lang/src/logger.rs:5]: oopsie

File: rudder-lang/src/opt.rs [hunk 2]
```diff
@@ -57,4 +57,8 @@ pub struct Opt {
     /// Use json logs instead of human readable output
     #[structopt(long, short)]
     pub json_log: bool,
-}
\ No newline at end of file
+
+    /// Generate a backtrace in case an error occurs
```
Comment (amousset) [inline rudder-lang/src/opt.rs:61]: Why is it an option? Do you think people might prefer without backtrace?

Comment (gpoblon) [reply rudder-lang/src/opt.rs:61]: Backtrace is useful for rudder-lang contributors, but end users won't really make use of it

**Comment by gpoblon** [comment]
add env_variable to have a backtrace on errors

**Comment by peckpeck** [comment]
rebase needed

**Comment by amousset** [comment]
the pom.xml commits should not be there

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Pull request merged**